### PR TITLE
CI improvements

### PR DESCRIPTION
**Pull request by dermetfan**
# Description

- run with exec driver instead of podman
  - more reliable
  - much better caching
  - faster
- update tullia
- only trigger on pushes to master so that CI is not triggered twice for every push to a PR

# Checklist

- Branch
    - [x] Commit sequence broadly makes sense
    - [x] Commits have useful messages
    - ~~The documentation has been properly updated~~
    - ~~New tests are added if needed and existing tests are updated~~
    - ~~Any changes affecting Consensus packages must have an entry in the appropriate `changelog.d` directory created using [`scriv`](https://github.com/input-output-hk/scriv). If in doubt, see the [Consensus release process](../ouroboros-consensus/docs/ReleaseProcess.md).~~
    - ~~If this branch changes Network and has any consequences for downstream repositories or end users, said changes must be documented in [`interface-CHANGELOG.md`](../docs/interface-CHANGELOG.md)~~
    - ~~If serialization changes, user-facing consequences (e.g. replay from genesis) are confirmed to be intentional.~~
- Pull Request
    - [x] Self-reviewed the diff
    - [x] Useful pull request description at least containing the following information:
      - What does this PR change?
      - Why these changes were needed?
      - ~~How does this affect downstream repositories and/or end-users?~~
      - ~~Which ticket does this PR close (if any)? If it does, is it [linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)?~~
    - [x] Reviewer requested

### CODE REVIEW

**Review by ce0la — APPROVED** [review]

File: nix/cells/cloud/actions.nix [hunk 1]
```diff
@@ -15,8 +15,8 @@
 
       #lib.merge
       #ios: [
-        #lib.io.github_push & github,
-        { #lib.io.github_pr, github, #target_default: false },
+        {#lib.io.github_push, github, #default_branch: true},
```
Comment (amesgen) [inline nix/cells/cloud/actions.nix:18]: We also need to build on Bors-related branches:

> Your CI system should build the “staging” and “trying” branches

https://bors.tech/documentation/getting-started/

See e.g. #4372 where Cicero did not run on the Bors commits.

**Comment by dermetfan** [comment]
bors try

File: nix/cells/cloud/actions.nix (hunk 1 above)
Comment (dermetfan) [reply nix/cells/cloud/actions.nix]: Ah thanks, fixed :slightly_smiling_face:

**Review by amesgen — APPROVED** [review]
Nice, uncached runs indeed seem to be much (~5x) faster :tada: 

Feel free to `bors merge` (potentially after squashing)

**Comment by dermetfan** [comment]
bors merge

**Pull request merged**